pkg/freeze/containerd/pause.go

```diff
@@ -40,46 +40,51 @@ func New() (*Containerd, error) {
 }
 
 // Freeze freezes the user container via the freezer cgroup.
-func (f *Containerd) Freeze(ctx context.Context, podName, containerName string) error {
+func (f *Containerd) Freeze(ctx context.Context, podName string) error {
 	ctrd, err := containerd.NewWithConn(f.conn)
 	if err != nil {
 		return err
 	}
 
-	containerID, err := lookupContainerID(ctx, f.conn, podName, containerName)
+	containers, err := f.GetContainers(ctx, f.conn, podName)
+	containerIDs, err := lookupContainerIDs(containers)
 	if err != nil {
 		return err
 	}
 
 	ctx = namespaces.WithNamespace(ctx, "k8s.io")
-	if _, err := ctrd.TaskService().Pause(ctx, &tasks.PauseTaskRequest{ContainerID: containerID}); err != nil {
-		return err
+	for _, c := range containerIDs {
+		if _, err := ctrd.TaskService().Pause(ctx, &tasks.PauseTaskRequest{ContainerID: c}); err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
 // Thaw thats a container which was freezed via the Freeze method.
-func (f *Containerd) Thaw(ctx context.Context, podName, containerName string) error {
+func (f *Containerd) Thaw(ctx context.Context, podName string) error {
 	ctrd, err := containerd.NewWithConn(f.conn)
 	if err != nil {
 		return err
 	}
 
-	containerID, err := lookupContainerID(ctx, f.conn, podName, containerName)
+	containers, err := f.GetContainers(ctx, f.conn, podName)
+	containerIDs, err := lookupContainerIDs(containers)
 	if err != nil {
 		return err
 	}
 
 	ctx = namespaces.WithNamespace(ctx, "k8s.io")
-	if _, err := ctrd.TaskService().Resume(ctx, &tasks.ResumeTaskRequest{ContainerID: containerID}); err != nil {
-		return err
+	for _, c := range containerIDs {
+		if _, err := ctrd.TaskService().Resume(ctx, &tasks.ResumeTaskRequest{ContainerID: c}); err != nil {
+			return err
+		}
 	}
-
 	return nil
 }
 
-func lookupContainerID(ctx context.Context, conn *grpc.ClientConn, podUID, containerName string) (string, error) {
+func (f *Containerd) GetContainers(ctx context.Context, conn *grpc.ClientConn, podUID string) (*cri.ListContainersResponse, error) {
```

Comment:
rather than exposing the private method just to test it, it'd be nice if we could test that Thaw and Freeze thaw and freeze the expected containers (i.e. by having a PauseResumer interface and checking that the right method is called with the expected containers). (Otherwise we're relying on Freeze and Thaw happening to call GetContainers and actually use the returned value etc, which isn't much of a test and won't survive very much refactoring)

Reply:
ok